# Ejercicio Tutorial_panadas_sesion1

### Eliana Bonalde

___

Ejercicio Final: 

Un médico requiere la siguiente información:

La media y la desviación estandar de la glucosa de los pacientes que fueron diagnosticasdos con diabétes y de los que fueron diagnosticados como no-diabéticos.

Posteriormente, quiere dividir a la población en grupos por rangos de edades de 10 años, es decir, de 0-10, de 10-20, etc. Y para cada uno de estos grupos, calcular la media y la desviación estandar de la glucosa.

In [3]:
import pandas as pd
import numpy as np

df = pd.read_csv('../2021_Pandas_tutorial/Data/diabetes.csv')
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

In [9]:
#df['Outcome']

In [8]:
df.groupby('Outcome').describe()

Pregnancies                                                Glucose  \
              count      mean       std  min   25%  50%  75%   max   count   
Outcome                                                                      
0             500.0  3.298000  3.017185  0.0  1.00  2.0  5.0  13.0   500.0   
1             268.0  4.865672  3.741239  0.0  1.75  4.0  8.0  17.0   268.0   

                     ... DiabetesPedigreeFunction           Age             \
               mean  ...                      75%    max  count       mean   
Outcome              ...                                                     
0        109.980000  ...                  0.56175  2.329  500.0  31.190000   
1        141.257463  ...                  0.72800  2.420  268.0  37.067164   

                                                  
               std   min   25%   50%   75%   max  
Outcome                                           
0        11.667655  21.0  23.0  27.0  37.0  81.0  
1        10.968254  21.0  28.0  36.0  44.0  70.0  

[2 rows x 64 columns]

La media de la glucosa de los pacientes que fueron diagnosticados como no-diabéticos es: 109.980000.
La media de la glucosa de los pacientes que fueron diagnosticados como diabéticos es: 141.257463.

___

Creamos la función `rango_edades` para agrupar por edades y la aplicamos a la columna 'Age'.

In [22]:
#df['Age'].unique()

In [15]:
def rango_edades(x):
    if x<=10:
        rango='rango_1'
    elif 10<x<=20:
        rango = 'rango_2'
    elif 20<x<=30:
        rango= 'rango_3'
    elif 30<x<=40:
        rango = 'rango_4'
    elif 40<x<=50:
        rango = 'rango_5'
    elif 50<x<=60:
        rango= 'rango_6'
    elif 60<x<=70:
        rango = 'rango_7'
    elif 70<x<=80:
        rango = 'rango_8'
    else:
        rango = 'rango_9'
        
    return rango   

In [18]:
df['RangoEdades'] = df.Age.apply(rango_edades) #aplicar la función rango_edades a la columna 'Age' 
df

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome RangoEdades  
0                       0.627   50        1     rango_5  
1                       0.351   31        0     rango_4  
2                       0.672   32        1     rango_4  
3                       0.167   21        0     rango_3  
4                       2.288   33        1     rango_4  
..                        ...  ...      ...         ...  
763                     0.171   63        0     rango_7  
764                     0.340   27        0     rango_3  
765                     0.245   30        0     rango_3  
766                     0.349   47        1     rango_5  
767                     0.315   23        0     rango_3  

[768 rows x 10 columns]

In [20]:
df.groupby('RangoEdades').agg(['mean','std'])

Pregnancies               Glucose            BloodPressure  \
                   mean       std        mean        std          mean   
RangoEdades                                                              
rango_3        2.007194  1.795412  114.175060  28.730772     65.323741   
rango_4        5.273885  3.197779  126.178344  33.375775     70.273885   
rango_5        7.123894  3.734809  124.884956  34.733874     74.787611   
rango_6        6.518519  3.374380  141.148148  31.338329     80.055556   
rango_7        4.800000  3.055050  137.560000  32.561583     78.080000   
rango_8        2.000000       NaN  119.000000        NaN      0.000000   
rango_9        9.000000       NaN  134.000000        NaN     74.000000   

                       SkinThickness                Insulin              \
                   std          mean        std        mean         std   
RangoEdades                                                               
rango_3      19.369341     21.796163  14.667277   84.306954  109.456203   
rango_4      19.908641     20.541401  16.841452   76.707006  112.185758   
rango_5      18.142517     19.389381  16.739300   60.274336  101.182054   
rango_6      10.298226     17.333333  15.930273  126.777778  181.455298   
rango_7      12.668596     11.920000  23.102886   14.800000   51.244512   
rango_8            NaN      0.000000        NaN    0.000000         NaN   
rango_9            NaN     33.000000        NaN   60.000000         NaN   

                   BMI           DiabetesPedigreeFunction            \
                  mean       std                     mean       std   
RangoEdades                                                           
rango_3      31.322062  8.691704                 0.446827  0.313057   
rango_4      32.886624  6.389616                 0.536994  0.374681   
rango_5      34.501770  5.769795                 0.452593  0.335631   
rango_6      31.114815  7.853695                 0.529111  0.351450   
rango_7      28.856000  7.583869                 0.430360  0.208961   
rango_8      19.600000       NaN                 0.832000       NaN   
rango_9      25.900000       NaN                 0.460000       NaN   

                   Age             Outcome            
                  mean       std      mean       std  
RangoEdades                                           
rango_3      24.599520  2.795996  0.215827  0.411889  
rango_4      35.146497  2.952216  0.484076  0.501346  
rango_5      44.327434  2.810829  0.566372  0.497783  
rango_6      54.981481  3.024995  0.574074  0.499126  
rango_7      64.880000  2.635021  0.280000  0.458258  
rango_8      72.000000       NaN  0.000000       NaN  
rango_9      81.000000       NaN  0.000000       NaN

No hay pacientes en rango_1 y rango_2.

Media y desviación estandar de la glucosa, respectivamente, de cada rango:
- rango_3: 114.175060 y 28.730772
- rango_4: 126.178344 y 33.375775
- rango_5: 124.884956 y 34.733874
- rango_6: 141.148148 y	31.338329	
- rango_7: 137.560000 y 32.561583
- rango_8: 119.000000 
- rango_9: 134.000000	